opensrp-app/src/test/java/org/smartregister/repository/dao/TaskDaoImplTest.java

```diff
@@ -151,4 +154,50 @@ public void testUpdateTaskShouldInvokeExpectedMethods() {
 
         ReflectionHelpers.setStaticField(LocalBroadcastManager.class, "mInstance", null);
     }
+
+    @Test
+    public void testSaveTaskShouldInvokeExpectedMethods() {
+        taskDao = Mockito.spy(taskDao);
+
+        LocalBroadcastManager localBroadcastManager = mock(LocalBroadcastManager.class);
+
+        doReturn(false).when(localBroadcastManager).sendBroadcast(any(Intent.class));
+
+        ReflectionHelpers.setStaticField(LocalBroadcastManager.class, "mInstance", localBroadcastManager);
+
+        org.smartregister.domain.Task task = new org.smartregister.domain.Task();
+        task.setIdentifier(UUID.randomUUID().toString());
+        task.setPriority(org.smartregister.domain.Task.TaskPriority.ROUTINE);
+        task.setStatus(ARCHIVED);
+
+        String locationId = "b8a7998c";
+
+        QuestionnaireResponse questionnaireResponse = QuestionnaireResponse.builder()
+                .item(QuestionnaireResponse.Item.builder()
+                        .linkId(com.ibm.fhir.model.type.String.of("location_id"))
+                        .definition(Uri.of("details"))
+                        .answer(QuestionnaireResponse.Item.Answer.builder().value(com.ibm.fhir.model.type.String.of(locationId)).build())
+                        .build())
+                .status(QuestionnaireResponseStatus.COMPLETED)
+                .build();
```

Comment:
Can we use sample event and convert to questionnaire response

Reply:
okay